Add LED support

davidwagnerkc opened the pull request:
We'll call this the pre-Derick designing crazy LED fanciness stage. 

Simple use of LEDs. 

Default - off. 

Wake word detected and listening - blue ring

Skill crunchin' some numbers and solving problems - blue/green rotating ring.

forslund commented on __init__.py:
```diff
@@ -273,40 +276,41 @@ def shutdown(self):
         handler = message.data.get('handler', '')
-        # Ignoring handlers from this skill and from the background clock
-        # TODO: implement Something here
+        if _skip_handler():
```
Same as above

forslund commented on __init__.py:
```diff
@@ -273,40 +276,41 @@ def shutdown(self):
+    def on_handler_started(self, message):
+        """When a skill begins executing turn on the LED ring"""
+        handler = message.data.get('handler', '')
+        if _skip_handler():
```
Should be `self._skip_handler()` maybe and there should probably be an argument as well

davidwagnerkc commented:
@forslund Can you take a second look?

I added LED indication for speaking per Derick's request, added some of his initial Mycroft color palette recommendations, and made it so that when volume is set or get that instead of the `speak()` LED indicator we display the volume on the LED ring. This volume feature requires another PR to work: https://github.com/MycroftAI/skill-volume/pull/44.

forslund commented:
Works fine. Merging
